plugins/application/elasticsearch/main.go

```diff
@@ -58,58 +33,122 @@ type Elasticsearch struct {
 	configuration *lib.AppConfig
 	logger        *logging.Logger
 	client        *lib.Client
-	store         chan esIndex
+	buffer        map[string][]string
+	dump          chan esIndex
 }
 
 //New constructor
 func New(logger *logging.Logger) application.Application {
 	return &Elasticsearch{
 		logger: logger,
-		store:  make(chan esIndex, 100),
+		buffer: make(map[string][]string),
+		dump:   make(chan esIndex, 100),
 	}
 }
 
+//getIndexName returns ES index name appropriate to data source and event type
+func getIndexName(record map[string]interface{}, source data.DataSource) string {
+	output := fmt.Sprintf("%s%s", source.String(), genericSuffix)
+
+	switch source.String() {
+	case "collectd":
```

Comment:
This is probably fine for now. I considered adding addressing in the internal buses so that application plugins could choose what addresses to subscribe to so that there is no risk of them getting the wrong message types. The address would be based on the handler name, such as "collectd-metrics," and the application would then only have to subscribe to that address and any additional that it wishes to handle. It would minimize the need for this switching logic. I will make that a todo but I think for now we should get this stuff merged

Reply:
Yup, some kind of addressing on internal bus make sense, definitely. Let me know if you would want to delegate that feature to somebody ;).